libs/community/langchain_community/vectorstores/sqlserver.py

```diff
@@ -746,33 +764,44 @@ def _insert_embeddings(
                     # For a query, if there is no corresponding ID,
                     # we generate a uuid and add it to the list of IDs to be returned.
                     if idx < len(ids):
-                        id = ids[idx]
+                        custom_id = ids[idx]
                     else:
                         ids.append(str(uuid.uuid4()))
-                        id = ids[-1]
+                        custom_id = ids[-1]
                     embedding = embeddings[idx]
                     metadata = metadatas[idx] if idx < len(metadatas) else {}
 
                     # Construct text, embedding, metadata as EmbeddingStore model
                     # to be inserted into the table.
-                    sqlquery = text(JSON_TO_ARRAY_QUERY).bindparams(
-                        bindparam(
-                            EMBEDDING_VALUES,
-                            json.dumps(embedding),
-                            # render the value of the parameter into SQL statement
-                            # at statement execution time
-                            literal_execute=True,
+                    sqlquery = select(
```

Comment:
what is the difference between doing a select call here v/s doing session.scalar(sqlquery)? Why choose one over the other?

Reply:
nvm re-read the description.